modules/core/02-client/migrations/v7/store.go

```diff
@@ -0,0 +1,215 @@
+package v7
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	"github.com/cosmos/cosmos-sdk/store/prefix"
+	storetypes "github.com/cosmos/cosmos-sdk/store/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+
+	clienttypes "github.com/cosmos/ibc-go/v6/modules/core/02-client/types"
+	host "github.com/cosmos/ibc-go/v6/modules/core/24-host"
+	"github.com/cosmos/ibc-go/v6/modules/core/exported"
+	solomachine "github.com/cosmos/ibc-go/v6/modules/light-clients/06-solomachine"
+	ibctm "github.com/cosmos/ibc-go/v6/modules/light-clients/07-tendermint"
+)
+
+// Localhost is the client type for a localhost client. It is also used as the clientID
+// for the localhost client.
+const Localhost string = "09-localhost"
+
+// MigrateStore performs in-place store migrations from ibc-go v6 to ibc-go v7.
+// The migration includes:
+//
+// - Migrating solo machine client states from v2 to v3 protobuf definition
+// - Pruning all solo machine consensus states
+// - Removing the localhost client
+// - Asserting existing tendermint clients are properly registered on the chain codec
+func MigrateStore(ctx sdk.Context, storeKey storetypes.StoreKey, cdc codec.BinaryCodec) error {
+	store := ctx.KVStore(storeKey)
+
+	if err := handleSolomachineMigration(ctx, store, cdc); err != nil {
+		return err
+	}
+
+	if err := handleTendermintMigration(ctx, store, cdc); err != nil {
+		return err
+	}
+
+	if err := handleLocalhostMigration(ctx, store, cdc); err != nil {
+		return err
+	}
+
+	return nil
+}
```

Comment:
very comprehensive doc string and super clean/simple implementation, I love it! 🥇